Gem/Code/Source/RHI/XRExampleComponent.cpp

```diff
@@ -55,35 +55,39 @@ namespace AtomSampleViewer
         if (xrSystem && xrSystem->ShouldRender())
         {
-            const AZ::RPI::FovData fovData = xrSystem->GetViewFov(m_viewIndex);
-            const AZ::RPI::PoseData poseData = xrSystem->GetViewPose(m_viewIndex);
+            AZ::RPI::FovData fovData;
+            AZ::RPI::PoseData poseData, frontViewPoseData;
+            [[maybe_unused]] AZ::RHI::ResultCode resultCode = xrSystem->GetViewFov(m_viewIndex, fovData);
+            resultCode = xrSystem->GetViewPose(m_viewIndex, poseData);
                 
             static const float clip_near = 0.05f;
             static const float clip_far = 100.0f;
+            bool reverseDepth = false;
             projection = xrSystem->CreateProjectionOffset(fovData.m_angleLeft, fovData.m_angleRight, 
                                                           fovData.m_angleDown, fovData.m_angleUp, 
-                                                          clip_near, clip_far);
+                                                          clip_near, clip_far, reverseDepth);
 
-            AZ::Quaternion poseOrientation = poseData.orientation; 
+            AZ::Quaternion poseOrientation = poseData.m_orientation; 
             poseOrientation.InvertFast(); 
-            AZ::Matrix4x4 viewMat = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(poseOrientation, -poseData.position);
+            AZ::Matrix4x4 viewMat = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(poseOrientation, -poseData.m_position);
             m_viewProjMatrix = projection * viewMat;
  
             const AZ::Matrix4x4 initialScaleMat = AZ::Matrix4x4::CreateScale(AZ::Vector3(0.1f, 0.1f, 0.1f));
 
             //Model matrix for the cube related to the front view
-            AZ::RPI::PoseData frontViewPoseData = xrSystem->GetViewFrontPose();
-            m_modelMatrices[0] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(frontViewPoseData.orientation, frontViewPoseData.position) * initialScaleMat;
+            resultCode = xrSystem->GetViewFrontPose(frontViewPoseData);
+            m_modelMatrices[0] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(frontViewPoseData.m_orientation, frontViewPoseData.m_position) * initialScaleMat;
                       
             //Model matrix for the cube related to the left controller
-            AZ::RPI::PoseData controllerLeftPose = xrSystem->GetControllerPose(0);
+            AZ::RPI::PoseData controllerLeftPose, controllerRightPose;
+            resultCode = xrSystem->GetControllerPose(0, controllerLeftPose);
             AZ::Matrix4x4 leftScaleMat = initialScaleMat * AZ::Matrix4x4::CreateScale(AZ::Vector3(xrSystem->GetControllerScale(0)));
-            m_modelMatrices[1] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(controllerLeftPose.orientation, controllerLeftPose.position) * leftScaleMat;
+            m_modelMatrices[1] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(controllerLeftPose.m_orientation, controllerLeftPose.m_position) * leftScaleMat;
```

Comment:
Would it make sense to make this a function of xrSystem?
For example: xrSystem->GetControllerTransform(controllderId)

Reply:
It may make sense later based on how we setup XR related components for usage within Editor + game launcher. At this point it is not clear how we want to render/update the controller. Currently I am expecting the calling code to figure out how to do that but if this changes in future we can create API to standardize hoe to render/update controller across O3de.